syft/frameworks/crypten/context.py

```diff
@@ -0,0 +1,152 @@
+import functools
+import multiprocessing
+import threading
+import os
+import crypten
+from crypten.communicator import DistributedCommunicator
+from syft.messaging.message import CryptenInit
+
+
+def _launch(func, rank, world_size, master_addr, master_port, queue, func_args, func_kwargs):
+    communicator_args = {
+        "RANK": rank,
+        "WORLD_SIZE": world_size,
+        "RENDEZVOUS": "env://",
+        "MASTER_ADDR": master_addr,
+        "MASTER_PORT": master_port,
+        "BACKEND": "gloo",
+    }
+    for key, val in communicator_args.items():
+        os.environ[key] = str(val)
+
+    crypten.init()
+    return_value = func(*func_args, **func_kwargs)
+    crypten.uninit()
+
+    queue.put(return_value)
+
+
+def _new_party(func, rank, world_size, master_addr, master_port, func_args, func_kwargs):
+    queue = multiprocessing.Queue()
+    process = multiprocessing.Process(
+        target=_launch,
+        args=(func, rank, world_size, master_addr, master_port, queue, func_args, func_kwargs),
+    )
+    return process, queue
+
+
+def run_party(func, rank, world_size, master_addr, master_port, func_args, func_kwargs):
+    """Start crypten party localy and run computation.
+
+    Args:
+        func (function): computation to be done.
+        rank (int): rank of the crypten party.
+        world_size (int): number of crypten parties involved in the computation.
+        master_addr (str): IP address of the master party (party with rank 0).
+        master_port (int or str): port of the master party (party with rank 0).
+        func_args (list): arguments to be passed to func.
+        func_kwargs (dict): keyword arguments to be passed to func.
+
+    Returns:
+        The return value of func.
+    """
+
+    process, queue = _new_party(
+        func, rank, world_size, master_addr, master_port, func_args, func_kwargs
+    )
+    was_initialized = DistributedCommunicator.is_initialized()
+    if was_initialized:
+        crypten.uninit()
+    process.start()
+    process.join()
+    if was_initialized:
+        crypten.init()
+    return queue.get()
+
+
+def _send_party_info(worker, rank, msg, return_values):
+    """Send message to worker with necessary information to run a crypten party.
+    Add response to return_values dictionary.
+
+    Args:
+        worker (BaseWorker): worker to send the message to.
+        rank (int): rank of the crypten party.
+        msg (CryptenInit): message containing the rank, world_size, master_addr and master_port.
+        return_values (dict): dictionnary holding return values of workers.
+    """
+
+    response = worker.send_msg(msg, worker)
+    return_values[rank] = response.contents
+
+
+def toy_func():
+    # Toy function to be called by each party
+    alice_t = crypten.cryptensor([73, 81], src=0)
+    bob_t = crypten.cryptensor([90, 100], src=1)
+    out = bob_t.get_plain_text()
```

Comment:
Can you do stuff like this?
```python
alice_t = crypten.cryptensor([73, 81], src=0)
bob_t = crypten.cryptensor([90, 100], src=1)
out = (alice_t + bob_t).get_plain_text()
```

Reply:
Just tested this locally to make sure. It works perfectly.

Reply:
```python
>>> import torch
>>> import crypten
>>> import syft
>>> from syft import WebsocketClientWorker
>>> from syft.frameworks.crypten.context import run_multiworkers
>>> 
>>> 
>>> hook = syft.TorchHook(torch)
>>> alice = WebsocketClientWorker(hook=hook, id="alice", host='127.0.0.1', port=8777)
>>> bob = syft.VirtualWorker(hook=hook, id="bob")
>>> 
>>> @run_multiworkers([alice, bob,], master_addr="127.0.0.1")
... def test():
...     pass
... 
>>> test()
{0: [163.0, 181.0], 1: [163.0, 181.0], 2: [163.0, 181.0]}